tools/uwuw_preproc

```diff
@@ -171,108 +118,184 @@ def tag_to_script(tag):
-                if mat_name[1] == '':
-                    raise Exception("Could not find group name in appropriate format; extra \'/\' in %s" % tag)
-                if ':' not in mat_name[1]:
-                    raise Exception("Could not find group name in appropriate format; ':' is absent after the '/' in %s" % tag)
-                matdensity = mat_name[1].split(':')
-                try:
-                    matdensity_test = float(matdensity[1])
-                except:
-                    raise Exception("Could not find density in appropriate format!; density is not a float in %s" % tag)
-                mat_list_density.append(matdensity[1])
+            if ('/' in tag):
+                splitted_group_name = mat_dens_split(tag)
             # otherwise we have only "mat:"
-            elif ':' in tag:
+            elif (':' in tag):
                 splitted_group_name = mat_split(tag)
             else:
                 raise Exception(
                     "Couldn\'t find group name in appropriate format; ': is absent' in  %s" % tag)
             mat_list_matname.append(splitted_group_name['material'])
             mat_list_density.append(splitted_group_name['density'])
-    if g == 0:
-        raise Exception("Graveyard group is missing! You must have a graveyard")
+    if (g == 0):
+        raise Exception(
+            "Graveyard group is missing! You must have a graveyard")
     mat_dens_list = zip(mat_list_matname, mat_list_density)
     # error conditions, no tags found
-#    if len(mat_dens_list) == 0:
-#        raise Exception("No material group names found, you must have materials")
+    if (len(mat_dens_list) == 0):
+        raise Exception(
+            "No material group names found, you must have materials")
 
     return mat_dens_list
 
 
 """
-splitting group names containing both materials and densities
+function that splits group name on the basis of '/'
+group name containing both material name and density
 """
 
 
 def mat_dens_split(tag):
     splitted_group_name = {}
     mat_name = tag.split('/')
-    if ':' not in mat_name[0]:
+    if (':' not in mat_name[0]):
         raise Exception(
             "Couldn\'t find group name in appropriate format; ':' is absent in %s" % tag)
     # list of material name only
     matname = mat_name[0].split(':')
-    if len(matname) > 2:
+    if (len(matname) > 2):
         raise Exception(
             "Wrong format for group names! %s. correct: mat:NAME/rho:VALUE or mat:NAME" % tag)
-    if matname[1] == '':
+    if (matname[1] == ''):
         raise Exception(
             "Couldn\'t find group name in appropriate format; wrong material name in %s" % tag)
     splitted_group_name['material'] = matname[1]
-    if mat_name[1] == '':
+    if (mat_name[1] == ''):
         raise Exception(
             "Couldn\'t find group name in appropriate format; extra \'/\' in %s" % tag)
```

Comment:
I think the fine-grain reporting of invalid tags is a bit excessive. Chances are you if you just raise an exception and print out the name of the offending tag, the user will have a clear idea of what is wrong.